From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
Date: Fri, 17 Oct 2025 15:01:00 +0530
Subject: [PATCH] Use const keyword for struct ctl_table for above 6.13 kernel
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

For struct ctl_table, const keyward has been added from 6.13 kernel to
protect its members.
Hence observing below warning.
To resolve this warning, not using const keyword for struct ctl_table for
below 6.13 kernels.

Reference:
	7abc9b53bd51 sysctl: allow registration of const struct ctl_table

Signed-off-by: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
---
 drivers/gpu/drm/xe/xe_observation.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_observation.c b/drivers/gpu/drm/xe/xe_observation.c
index e3f9b54..e47d7c9 100644
--- a/drivers/gpu/drm/xe/xe_observation.c
+++ b/drivers/gpu/drm/xe/xe_observation.c
@@ -70,7 +70,11 @@ int xe_observation_ioctl(struct drm_device *dev, void *data, struct drm_file *fi
 	}
 }
 
+#ifdef BPM_STRUCT_CTL_TABLE_CONST_KEYWORD_NOT_PRESENT
+static struct ctl_table observation_ctl_table[] = {
+#else
 static const struct ctl_table observation_ctl_table[] = {
+#endif
 	{
 	 .procname = "observation_paranoid",
 	 .data = &xe_observation_paranoid,
-- 
2.43.0

